--- acpica-unix-20170303/source/include/platform/aclinux.h.orig	2017-06-12 22:05:52.100041265 +0200
+++ acpica-unix-20170303/source/include/platform/aclinux.h	2017-06-12 22:06:02.790569724 +0200
@@ -310,7 +310,7 @@
 #define ACPI_FLUSH_CPU_CACHE()
 #define ACPI_CAST_PTHREAD_T(Pthread) ((ACPI_THREAD_ID) (Pthread))
 
-#if defined(__ia64__)    || defined(__x86_64__) ||\
+#if defined(__ia64__)    || (defined(__x86_64__) && !defined(__ILP32__)) ||\
     defined(__aarch64__) || defined(__PPC64__) ||\
     defined(__s390x__)
 #define ACPI_MACHINE_WIDTH          64
