diff -Naur acpica-unix2-20211217/source/components/utilities/utdebug.c acpica-unix2-20211217.orig/source/components/utilities/utdebug.c
--- acpica-unix2-20211217/source/components/utilities/utdebug.c	2022-03-15 17:05:07.992387565 -0600
+++ acpica-unix2-20211217.orig/source/components/utilities/utdebug.c	2021-12-17 10:48:16.000000000 -0700
@@ -58,6 +58,10 @@
 static const char           *AcpiGbl_FunctionExitPrefix  = "----Exit-";
 
 
+#if defined(__GNUC__) && (__GNUC__ > 11)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdangling-pointer"
+#endif
 /*******************************************************************************
  *
  * FUNCTION:    AcpiUtInitStackPtrTrace
@@ -114,6 +110,9 @@
         AcpiGbl_DeepestNesting = AcpiGbl_NestingLevel;
     }
 }
+#if defined(__GNUC__) && (__GNUC__ > 11)
+#pragma GCC diagnostic pop
+#endif
 
 
 /*******************************************************************************
