Index: acpica-unix-20180508/source/compiler/dttable2.c
===================================================================
--- acpica-unix-20181031/source/compiler/dttable2.c.orig	2018-11-18 21:47:12.950401024 +0100
+++ acpica-unix-20181031/source/compiler/dttable2.c	2018-11-20 05:42:52.529088619 +0100
@@ -1962,6 +1962,7 @@ DtCompileTpm2 (
     DT_SUBTABLE             *ParentTable;
     ACPI_STATUS             Status = AE_OK;
     ACPI_TABLE_HEADER       *Header;
+    UINT32		    Tmp32;
 
 
     ParentTable = DtPeekSubtable ();
@@ -2005,7 +2006,8 @@ DtCompileTpm2 (
 
     /* Subtable type depends on the StartMethod */
 
-    switch (Tpm2Header->StartMethod)
+    ACPI_MOVE_32_TO_32(&Tmp32, &Tpm2Header->StartMethod);
+    switch (Tmp32)
     {
     case ACPI_TPM2_COMMAND_BUFFER_WITH_ARM_SMC:
 
@@ -2035,16 +2037,14 @@ DtCompileTpm2 (
     case ACPI_TPM2_RESERVED9:
     case ACPI_TPM2_RESERVED10:
 
-        AcpiOsPrintf ("\n**** Reserved TPM2 Start Method type 0x%X\n",
-            Tpm2Header->StartMethod);
+        AcpiOsPrintf ("\n**** Reserved TPM2 Start Method type 0x%X\n", Tmp32);
         Status = AE_ERROR;
         break;
 
     case ACPI_TPM2_NOT_ALLOWED:
     default:
 
-        AcpiOsPrintf ("\n**** Unknown TPM2 Start Method type 0x%X\n",
-            Tpm2Header->StartMethod);
+        AcpiOsPrintf ("\n**** Unknown TPM2 Start Method type 0x%X\n", Tmp32);
         Status = AE_ERROR;
         break;
     }
