Index: acpica-unix2-20220331/source/compiler/aslparseop.c
===================================================================
--- acpica-unix2-20220331.orig/source/compiler/aslparseop.c
+++ acpica-unix2-20220331/source/compiler/aslparseop.c
@@ -285,7 +285,16 @@ TrCreateValuedLeafOp (
 
 
     Op = TrAllocateOp (ParseOpcode);
-    Op->Asl.Value.Integer = Value;
+    if (ParseOpcode == PARSEOP_NAMESTRING ||
+        ParseOpcode == PARSEOP_NAMESEG ||
+        ParseOpcode == PARSEOP_STRING_LITERAL)
+    {
+        Op->Asl.Value.String = (char *) Value;
+    }
+    else
+    {
+        Op->Asl.Value.Integer = Value;
+    }
 
     DbgPrint (ASL_PARSE_OUTPUT,
         "\nCreateValuedLeafOp  Ln/Col %u/%u NewOp %p  "
