diff -Naur acpica-unix2-20170119.orig/source/include/platform/aclinux.h acpica-unix2-20170119/source/include/platform/aclinux.h
--- acpica-unix2-20170119.orig/source/include/platform/aclinux.h	2017-01-30 15:38:53.465753721 -0700
+++ acpica-unix2-20170119/source/include/platform/aclinux.h	2017-01-30 15:42:55.911358257 -0700
@@ -219,9 +219,11 @@
 #define __cdecl
 #endif
 
+#if defined(__BIG_ENDIAN__)
 #if defined(__PPC64__) || defined(__s390x__)
 #define ACPI_BIG_ENDIAN
 #endif
+#endif
 
 #endif /* __KERNEL__ */
 
